tensorflow/core/framework/op_kernel.h

```diff
@@ -169,6 +169,7 @@ class OpKernel {
     return output_memory_types_;
   }
 
+  bool HasInput(StringPiece input_name) const;
```

Comment:
Revert the changes to these files?

Reply:
I got the suggestion from @protoget to add this checking here. Which one should I follow?

Reply:
I'm a lot more comfortable approving changes which do not touch core tf functionality like op_kernel.{h,cc}.

Since a quick search over the non-contrib part of your pull request didn't find any usages of these new symbols I think it's more prudent to drop them.

Reply:
Sure. I reverted this changes on op_kernel in the new commit.